src/load.c

```diff
@@ -277,6 +309,83 @@ static int load_enumerateCallback( void* data, const char* origdir, const char*
    return PHYSFS_ENUM_OK;
 }
 
+/**
+ * @brief The PHYSFS_EnumerateCallback for load_refreshPlayerNames
+ */
+static int load_enumeratePlayerNamesCallback( void* data, const char* origdir, const char* fname )
+{
+   char *path, *backup_path;
+   const char *fmt;
+   size_t dir_len, name_len;
+   filedata_t *tmp;
+   PHYSFS_Stat stat;
+
+   dir_len = strlen( origdir );
+   name_len = strlen( fname );
+
+   fmt = dir_len && origdir[dir_len-1]=='/' ? "%s%s" : "%s/%s";
+   asprintf( &path, fmt, origdir, fname );
+   if (!PHYSFS_stat( path, &stat ))
+      WARN( _("PhysicsFS: Cannot stat %s: %s"), path,
+            PHYSFS_getErrorByCode( PHYSFS_getLastErrorCode() ) );
+   else if (stat.filetype == PHYSFS_FILETYPE_REGULAR) {
+      if ((name_len < 4 || strcmp( &fname[name_len-3], ".ns" )) && (name_len < 11 || strcmp( &fname[name_len-10], ".ns.backup" ))) {
+         free( path );
+         return PHYSFS_ENUM_OK;
+      }
+      if (!PHYSFS_exists( "saves-pre-0.10.0" ))
+         PHYSFS_mkdir( "saves-pre-0.10.0" );
+      asprintf( &backup_path, "saves-pre-0.10.0/%s", fname );
+      if (!ndata_copyIfExists( path, backup_path ))
+         old_saves_detected = 1;
+      free( backup_path );
+      if (name_len >= 4 && !strcmp( &fname[name_len-3], ".ns" )) {
```

Comment:
You can probably consolidate the block here and the next one for `.ns.backup` into an external function to reduce code duplication.